src/shared/format-table.c

```diff
@@ -1069,6 +1087,37 @@ int table_set_sort(Table *t, size_t first_column, ...) {
         return 0;
 }
 
+int table_hide_column_from_display(Table *t, size_t column) {
+        size_t allocated, cur = 0;
+        int r;
+
+        assert(t);
+        assert(column < t->n_columns);
+
+        /* If the display map is empty, initialize it with all available columns */
+        if (!t->display_map) {
+                r = table_set_display_all(t);
+                if (r < 0)
+                        return r;
+        }
+
+        allocated = t->n_display_map;
+
+        for (size_t i = 0; i < allocated; i++) {
+                if (t->display_map[i] == column)
+                        continue;
+
+                t->display_map[cur++] = t->display_map[i];
+        }
+
+        if (!GREEDY_REALLOC(t->display_map, allocated, cur))
```

Comment:
This needs to be `cur + 1`.

Reply:
Whoops, thanks!

Reply:
Duh, sorry, my comment was wrong. `GREEDY_REALLOC` never shrinks, so those two lines should be removed (they don't do anything).

Reply:
Ah, interesting, thank you.

Reply:
That actually explains why valgrind/ASan/UBsan didn't catch the off-by-one above.

Reply:
Yep. It is possible that the allocated area is bigger than what we "think" based on n_display_map, but that doesn't cause any issues, even if we reallocate later, since glibc maintains its own information about the allocated size.